src/main.js

```diff
@@ -1,12 +1,103 @@
 import React from 'react';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import beasts from './data.json';
 import {HornedBeasts} from './hornedBeasts';
-import './main.css';
 
+
+/**
+ *
+ * @returns {JSX.Element}
+ * @constructor
+ */
 export function Main() {
+  'use strict';
+
   return (
     <main>
-      <HornedBeasts description="A goat" title="Hi" imageUrl="./logo192.png" altText="It says Hi"/>
-      <HornedBeasts description="An elk" title="How are you?" imageUrl="./logo192.png" altText="It says How are you?"/>
+      {beastMaker()}
     </main>
   );
+
+  /**
+   *
+   * @param {number} columns
+   * @returns {JSX.Element[]}
+   */
+  function beastMaker(columns = 3) {
+    /*
+    const result = [];
+
+    for (let i = 0; i < beasts.length; i += columns) {
+      let row = document.createElement('Row');
+      for (let j = 0; j < columns; j++) {
+        try {
+          row.appendChild(<HornedBeasts description={beasts[i + j].description}
+                                        title={beasts[i + j].title}
+                                        imageUrl={beasts[i + j].image_url}
+                                        altText={beasts[i + j].keyword}/>);
+        } catch (e) {
+          console.log(e);
+        }
+      }
+      result.push(row);
+    }
+    return result;
+     */
+
+    // I'd like to know how to do this more programmatically
+    // like the above attempt, with a Row reference,
+    // appending children to it before pushing to the array
+

```

Comment:
Hey Joe, 
 I really like how you're thinking here!

I would ask a lab TA or perhaps JB to see if they can help you find a working solution should you want to continue on this path.

You may need to use `React.createElement` instead of `document.createElement`, and there may be off-by-one errors in the loops, but you're on to something here for sure :)  I know Bootstrap code can get quite repetitive.

I haven't used Bootstrap in quite sometime, but I will say that the React Bootstrap Grid system is still a good approach as well here.

https://react-bootstrap.github.io/layout/grid/